### PR TITLE
FOIA-281 Report section 2 data type filter

### PR DESCRIPTION
**Pull request by pixlkat**
This adds a bit more functionality than the ticket asked for. Only the data type "Requests" has filter options available for the modal. the rest of them will now just be empty select fields. The others will get added in whatever ticket is marked to fix up the modal functionality. One thing I noticed is that i am unable to type in the value field. I also didn't touch the random buttons at the bottom.

### CODE REVIEW

File: js/pages/annual_report_data.jsx [hunk 1]
```diff
@@ -7,11 +7,12 @@ import annualReportDataFormStore from '../stores/annual_report_data_form';
 import FoiaReportFormSectionThree from '../components/foia_report_form_section_three';
 import agencyComponentStore from '../stores/agency_component';
 import annualReportFiscalYearStore from '../stores/annual_report_fiscal_year';
+import annualReportDataTypesStore from '../stores/annual_report_data_types';
 import { reportActions } from '../actions/report';
 
 class AnnualReportDataPage extends Component {
   static getStores() {
-    return [annualReportDataFormStore, annualReportFiscalYearStore, agencyComponentStore];
+    return [annualReportDataFormStore, annualReportFiscalYearStore, agencyComponentStore, annualReportDataTypesStore];
```
Comment (bemarlan) [inline js/pages/annual_report_data.jsx]: `make test` is complaining about the line length. Can you break this up between lines?

File: js/stores/annual_report_data_form.js [hunk 2]
```diff
@@ -56,6 +56,40 @@ class AnnualReportDataFormStore extends Store {
         this.__emitChange();
         break;
       }
+
+      case types.ANNUAL_REPORT_DATA_TYPE_UPDATE:
```
Comment (bemarlan) [inline js/stores/annual_report_data_form.js]: Linting doesn't want any const to be declared inside the case like this. Example on moving const outside: https://eslint.org/docs/rules/no-case-declarations

File: js/stores/annual_report_data_form.js [hunk 3]
```diff
@@ -56,6 +56,40 @@ class AnnualReportDataFormStore extends Store {
...
+        selectedDataTypes.splice(previousDataType.index, 1, selectedDataType);
+        Object.assign(this.state, {
+          selectedDataTypes,
+        });
+
+        this.__emitChange();
+        break;
+
+      case types.ANNUAL_REPORT_DATA_TYPE_FILTER_ADD_GROUP: {
+        const selectedDataTypes = [...this.state.selectedDataTypes];
```
Comment (bemarlan) [inline js/stores/annual_report_data_form.js]: This name is already declared on line 12 so linting is complaining. Maybe just use a different const name.

File: js/components/foia_report_form_section_two.jsx [hunk 4]
```diff
@@ -1,101 +1,69 @@
-import React, { Component } from 'react';
+import React, {Component} from 'react';
```
[the remark this answers is not in the document]
Comment (pixlkat) [reply js/components/foia_report_form_section_two.jsx]: oops. thought I'd caught all those.

File: js/stores/annual_report_data_form.js (hunk 2 above)
Comment (pixlkat) [reply js/stores/annual_report_data_form.js]: ah. forgot the curly braces. I hate javascript.

File: js/stores/annual_report_data_form.js (hunk 3 above)
Comment (pixlkat) [reply js/stores/annual_report_data_form.js]: fixed. I see it is also complaining about duplicated code elsewhere, but the code isn't actually a duplicate.

**Pull request merged**